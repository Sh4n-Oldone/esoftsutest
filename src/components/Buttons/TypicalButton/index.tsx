import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

const TypicalButton: React.FC<{
  action: () => void;
  text: string;
  buttonStyles?: {} | {}[];
  textStyles?: {} | {}[];
}> = ({action, text, buttonStyles, textStyles}) => {
  return (
    <TouchableOpacity onPress={action} style={buttonStyles || styles.button}>
      <Text style={textStyles || styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#515ee9',
    paddingVertical: 6,
    paddingHorizontal: 10,
    minWidth: 100,
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
  },
});

export default TypicalButton;
