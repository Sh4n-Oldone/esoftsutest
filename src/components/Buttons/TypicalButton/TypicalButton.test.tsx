import 'react-native';
import React from 'react';
import TypicalButton from './index';

import renderer from 'react-test-renderer';

describe('Correctly rendered default button', () => {
  it('NavigationButton with 2 props renders correctly', () => {
    const btn = renderer.create(
      <TypicalButton action={() => {}} text={'bttn'} />,
    );
    expect(btn.toJSON()).toMatchSnapshot();
    expect(btn.root.findByType('Text').children).toContain('bttn');
  });

  it('NavigationButton with 4 props renders correctly', () => {
    const btn = renderer.create(
      <TypicalButton
        action={() => {}}
        text={'bttn'}
        textStyles={{color: '#000'}}
        buttonStyles={{backgroundColor: '#000'}}
      />,
    );
    expect(btn.toJSON()).toMatchSnapshot();
    expect(btn.root.findByType('Text').children).toContain('bttn');
    expect(btn.toJSON().props.style.backgroundColor).toBe('#000');
    expect(btn.root.findByType('Text').props.style).toEqual({
      color: '#000',
    });
  });
});
