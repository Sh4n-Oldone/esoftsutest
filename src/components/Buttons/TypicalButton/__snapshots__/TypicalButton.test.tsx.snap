// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered default button NavigationButton with 2 props renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#515ee9",
      "minWidth": 100,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 6,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
    bttn
  </Text>
</View>
`;

exports[`Correctly rendered default button NavigationButton with 4 props renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#000",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
      }
    }
  >
    bttn
  </Text>
</View>
`;
